### PR TITLE
Earth - Alice B.

### PR DESCRIPTION
**Pull request by aliceboone**
# Task List
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
:------------- | :-------------
Describe in your own words what the Model is doing in Rails | It is a class that can add, find, update or remove data, the methods refer to the  CRUD--Create, Remove, Update, Destroy.

Describe in your own words what the Controller is doing in Rails | In Rails the controller is responsible for making the request and produce the right output

Describe in your own words what the View is doing in Rails |  Each controller has directory in the app/views directory which holds the template files that have the views, these files are used to display the view that results from the respective controller action.
Describe an edge-case controller test you wrote |When update or delete a task will redirect to the root page if given an invalid id or will responde  with not found for invalid id.

What is the purpose of using strong params? (i.e. the params method in the controller) | The params is accessed by the application via ApplicationController, it refers to the parameters being passed to the controller via a GET or POST request.

How are Rails migrations related to Rails models? | Migrations are a convenient way to change the database schema it can means that each migration as being a new 'version' of the database.

Describe one area of Rails that are still unclear on | Some stuff is kinda confused yet, I was able to do my exercise following the videos and classes, but sure I have a lot to deep learning yet.

### CODE REVIEW

File: app/views/tasks/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+      <%= f.text_field :name %>
+    </div>
+    <div class="label">
+      <%= f.label :description %>&nbsp&nbsp&nbsp&nbsp
+      <%= f.text_area :description %>
+    </div>
+    <div class="label">
+      <label for="start">Completed_at</label>
+      <input type="date" id="start" name="trip-start"
+    </div>
```
Comment (dHelmgren) [inline app/views/tasks/new.html.erb:15]: by putting this here, rather than leaving it empty, you end up setting the task completed every time you make a new one.

**Review by dHelmgren — COMMENTED** [review]
# Task List




## Major Learning Goals/Code Review





| Criteria                                           | yes/no, and optionally any details/lines of code to reference |
| -------------------------------------------------- | ------------------------------------------------------------- |
| At least 6 commits with meaningful commit messages | ✔️                                                            |
| Routes follow RESTful conventions                  | ✔️                                                            |
| Uses named routes (like `_path`)                   | ✔️                                                            |
| Creates Models and migrations                      | ✔️                                                            |
| Creates styled views                               | ✔️                                                            |
| Handles errors like nonexistant tasks              | ✔️                                                            |
| Uses `form_with` to render forms in Rails          | ✔️                                                            |

## Functional Requirements/Manual Testing

| Functional Requirement                                                         | yes/no |
| ------------------------------------------------------------------------------ | ------ |
| Successfully handles index & show                                              | ✔️     |
| index & show tests pass                                                        | ✔️     |
| Successfully handles: New, Create                                              | see comment     |
| New, Create tests pass                                                         | ✔️     |
| Successfully handles: Edit, Update                                             | ✔️     |
| Edit, Update tests pass with valid & invalid task ids                          | ✔️     |
| Successfully handles: Destroy, Task Complete                                   | ✔️     |
| Tests for Destroy & Task Complete include tests for valid and invalid task ids | ✔️     |

## Overall Feedback

| Overall Feedback                | Criteria                                                                                                                                                                                      | yes/no |
| ------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------ |
| Green (Meets/Exceeds Standards) | 5+ in Code Review && 6+ in Functional Requirements                                                                                                                                            | ✔️
| Yellow (Approaches Standards)   | 3+ in Code Review && 5+ in Functional Requirements, or the instructor judges that this project needs special attention                                                                        |
| Red (Not at Standard)           | 0-2 in Code Review or 0-4 in Functional Reqs, or assignment is breaking/doesn’t run with less than 5 minutes of debugging, or the instructor judges that this project needs special attention |


Good job overall! One thing to watch out for is making plenty of regular commits! **One commit per wave is going to come back to bite you in the coming weeks!** Finally, take a quick peek at the comment in your code, there is an error in the way your data gets set up.
